liota/dccs/iotcc.py

```diff
@@ -513,8 +540,16 @@ def store_reg_entity_attributes(self, entity_type, entity_name, reg_entity_id,
                 new_prop_dict = tmp_dict
         else:
             tmp_dict = self.read_entity_file(reg_entity_id)
-            if ((tmp_dict["entity type"] == entity_type) and (tmp_dict["name"] == entity_name)
-                and (tmp_dict["device type"] == dev_type)):
+            # check Entity_Timestamp of entity_file: if < boottime, get properties from cloud
+            if (self.enable_reboot_getprop == "True") and ('Entity_Timestamp' in tmp_dict):
+                last_dtime = datetime.datetime.strptime(tmp_dict["Entity_Timestamp"], "%Y-%m-%dT%H:%M:%S")
+                if (last_dtime <= self.boottime):
+                    list_prop = self.get_properties(reg_entity_id)
```

Comment:
If i understand correctly, store_reg_entity_attributes() gets called for every set_properties() call. So if the gateway reboots and a user package has made 20 set_properties() call, your code will call get_properties() every time set_properties() call is made? Isn't that too many calls for the same entity?

What if user had set properties using a package, which is not to be loaded after reboot? Get_properties call will never be made?

Reply:
store_reg_entity_attributes() will be called not only when you call set_properties(), but when you call register().
In liota package, we always have to call register() to create or find the resources in HelixAdapter. Therefore, after reboot, get_properties() can at least be called once.

In fact, it will only be called ONCE, even you have 20 set_properties(). The reason is that whenever I modify the local record under entity folder (configurable), I will put the current time to "Entity_Timestamp". 
Then, Line 543-546 can guarantee get_properties will not be called any more, unless you have another reboot.

Reply:
Liota package is a user code. We cannot force a user to always call register() or set_properties(). You can not assume that get_properties() will be called.

Reply:
If no register() nor set_properties() are called, AW will never get a uuid.json file to take. Nothing will be impacted. I did not assume user to call register() or set_properties(). But those are the only chance AW agent can get a new uuid.json file to take. When that happens, if reboot had occurred,  get_properties() will be called to make AW agent to have a chance to have updated/synced record of properties from voprs.